network.go

```diff
@@ -685,6 +686,14 @@ func (n *network) CreateEndpoint(name string, options ...EndpointOption) (Endpoi
 		}
 	}()
 
+	// Watch for service records
+	n.getController().watchSvcRecord(ep)
```

Comment:
No, it should be there. This is to update the localEps in the watchDb with a version of endpoint which has sandbox information since we are triggering the watch during create endpoint.  But this will not result in the endpoint getting added twice in /etc/hosts